src/libraries/System.Text.Json/gen/Helpers/RoslynExtensions.cs

```diff
@@ -269,5 +278,107 @@ public static string GetTypeKindKeyword(this TypeDeclarationSyntax typeDeclarati
                     return null;
             }
         }
+
+        public static void ResolveNullabilityAnnotations(this IFieldSymbol field, out bool isGetterNonNullable, out bool isSetterNonNullable)
```

Comment:
In this case we want the source generator model to identify members that has nullable types with a non-nullable annotation, e.g. `string` but not `int` (see comments on `PropertyGenerationSpec`). This is because we don't want the fast-path serialization method emitter to include superfluous null checks for the latter case.

The two properties are equivalent assuming that you know if a given type is a reference type or `Nullable<T>`, however symbol information isn't available at the emitter stage of the source generator, so it is more convenient represent the model in this form.